llvm/lib/Transforms/Utils/SimplifyLibCalls.cpp

```diff
@@ -2589,10 +2584,26 @@ Value *LibCallSimplifier::optimizeLog(CallInst *Log, IRBuilderBase &B) {
       PowLb = LibFunc_powl;
       break;
     default:
-      return Ret;
+      return nullptr;
+    }
+
+    // Convert libcall to intrinsic if the value is known > 0.
+    bool IsKnownNoErrno = Log->hasNoNaNs() && Log->hasNoInfs();
+    if (!IsKnownNoErrno) {
+      SimplifyQuery SQ(DL, TLI, DT, AC, Log, true, true, DC);
+      KnownFPClass Known = computeKnownFPClass(
+          Log->getOperand(0), KnownFPClass::OrderedLessThanZeroMask,
+          /*Depth=*/0, SQ);
+      IsKnownNoErrno =
+          Known.isKnownNeverZero() && Known.cannotBeOrderedLessThanZero();
```

Comment:
Broken for DAZ, you want isKnownNeverLogicalZero (we should probably remove isKnownNeverZero)

Reply:
I had given denormals a test with log and in the routine I was using they had not been treated as zero (unlike fmod where they were). I've changed this over though.

Reply:
It entirely depends on what happens in the implementation, you have to conservatively assume denormals may be treated as zeroes.